tls/extensions/s2n_client_supported_versions.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <sys/param.h>
+#include <stdint.h>
+
+#include "tls/s2n_alerts.h"
+#include "tls/s2n_tls_parameters.h"
+#include "tls/s2n_tls.h"
+#include "tls/extensions/s2n_client_supported_versions.h"
+
+#include "utils/s2n_safety.h"
+
+/**
+ * Specified in https://tools.ietf.org/html/rfc8446#section-4.2.1
+ *
+ * "The "supported_versions" extension is used by the client to indicate
+ * which versions of TLS it supports and by the server to indicate which
+ * version it is using. The extension contains a list of supported
+ * versions in preference order, with the most preferred version first."
+ **/
+
+uint8_t s2n_supported_protocol_versions[] = { S2N_TLS13, S2N_TLS12, S2N_TLS11, S2N_TLS10, S2N_SSLv3, S2N_SSLv2 };
```

Comment:
Should we include SSLv2 here? I don't think we really support it ... just SSLv2 compatible hello messages, but I don't think that's what this is about.

Reply:
We probably shouldn't. We should probably actually only use the versions we would previously accept in a client hello, which means only the versions > cipher_preferences->minimum_protocol_version.